src/main/java/core/basesyntax/strategy/SupplyOperation.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.ShopOperation;
+import exception.OperationException;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(ShopOperation shopOperation) {
+        try {
+            String fruit = shopOperation.getFruit();
+            int quantity = Storage.fruitsStorage.get(fruit);
+            Storage.fruitsStorage.put(fruit, quantity + shopOperation.getQuantity());
+        } catch (NullPointerException e) {
+            throw new OperationException("Operation is not correct: " + shopOperation);
+        }
```

Comment:
Catching `NullPointerException` is generally not a best practice for control flow. Instead, consider checking if the fruit exists in the storage before attempting to retrieve its quantity. This can prevent the exception from occurring in the first place.